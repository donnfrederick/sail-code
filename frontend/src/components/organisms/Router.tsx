import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
