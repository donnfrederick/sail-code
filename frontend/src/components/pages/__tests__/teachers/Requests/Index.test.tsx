import Router from 'components/organisms/Router'
import RequestsIndex from 'components/pages/teachers/Requests/Index'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <RequestsIndex />', () => {
  const tree = renderer
    .create(
      <Router>
        <RequestsIndex />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
