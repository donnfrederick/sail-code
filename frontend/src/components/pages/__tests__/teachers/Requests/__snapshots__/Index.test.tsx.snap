// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <RequestsIndex /> 1`] = `
<div
  className="sc-iRbamj hdvZhl"
>
  <div
    className="sc-gPEVay kvMfKm"
  />
  <div
    className="sc-EHOje itwipb"
  >
    <div
      className="sc-bZQynM bZKBNj"
    >
      予約リクエスト一覧
    </div>
    <a
      className="sc-gzVnrw jTRbiO"
      href="/teachers/mypage"
      onClick={[Function]}
    >
      ホームに戻る
    </a>
    <div
      className="sc-htoDjs fUCPvq"
    >
      <a
        className="sc-dnqmqq engvPe"
        href="/teachers/notifications"
        onClick={[Function]}
      />
      <a
        className="sc-iwsKbI jCgDjX"
        href="/teachers/support"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
